pkg/distributor/forwarding/forwarding.go

```diff
@@ -515,7 +515,9 @@ func (r *request) doHTTP(ctx context.Context, body []byte) error {
 	httpReq.Header.Set(user.OrgIDHeaderName, r.user)
 
 	r.requests.Inc()
-	r.samples.Add(float64(r.counts.SampleCount))
+	if r.counts.SampleCount > 0 {
```

Comment:
Nit: these `if` checks before increasing metrics are not necessary. I'd remove them, unless you can show significant improvement by introducing them.

Reply:
ack, removing before "Add" calls.